accounts/abi/abi.go

```diff
@@ -117,11 +118,80 @@ func (abi ABI) Pack(name string, args ...interface{}) ([]byte, error) {
 	return append(method.Id(), arguments...), nil
 }
 
+// toGoSliceType prses the input and casts it to the proper slice defined by the ABI
+// argument in T.
+func toGoSlice(i int, t Argument, output []byte) (interface{}, error) {
+	index := i * 32
+	// The slice must, at very least be large enough for the index+32 which is exactly the size required
+	// for the [offset in output, size of array]. The size of the array (in elements) is exactly 32 bytes.
+	if index+32 > len(output) {
+		return nil, fmt.Errorf("abi: cannot marshal in to go slice: insufficient size output %d require %d", len(output), index+32)
+	}
+
+	// first we need to create a slice of the type
+	var refSlice reflect.Value
+	switch t.Type.T {
+	case IntTy, UintTy, BoolTy: // int, uint, bool can all be of type big int.
+		refSlice = reflect.ValueOf([]*big.Int(nil))
+	case AddressTy: // address must be of slice Address
+		refSlice = reflect.ValueOf([]common.Address(nil))
+	case HashTy: // hash must be of slice hash
+		refSlice = reflect.ValueOf([]common.Hash(nil))
+	default: // no other types are supported
+		return nil, fmt.Errorf("abi: unsupported slice type %v", t.Type.T)
+	}
+	// get the offset which determines the start of this array ...
+	offset := int(common.BytesToBig(output[index : index+32]).Uint64())
+	slice := output[offset:]
+	// ... starting with the size of the array in elements ...
+	size := int(common.BytesToBig(slice[:32]).Uint64())
```

Comment:
Size needs to be validated to since it's not possible to verify the buffer's required size prior to parsing the size value here.